modules/openapi-generator/src/main/resources/swift-combine/api.mustache

```diff
@@ -0,0 +1,215 @@
+{{#operations}}//
+// {{classname}}.swift
+//
+// Generated by openapi-generator
+// https://openapi-generator.tech
+
+import Foundation
+import Combine
+import OpenAPITransport
+
+{{#description}}
+/// {{{.}}} {{/description}}
+open class {{classname}} {
+    private let transport: OpenAPITransport
+    public var encoder: JSONEncoder = {
+        let encoder = JSONEncoder()
+        encoder.dateEncodingStrategy = .formatted(OpenISO8601DateFormatter())
+        return encoder
+    }()
+    public var decoder: JSONDecoder = {
+        let decoder = JSONDecoder()
+        decoder.dateDecodingStrategy = .formatted(OpenISO8601DateFormatter())
+        return decoder
+    }()
+    public var baseURL = URL(string: "{{{basePath}}}")
+
+    public init(_ transport: OpenAPITransport) {
+        self.transport = transport
+    }
+{{#operation}}
+
+    {{#allParams}}
+    {{#isEnum}}
+    ///
+    /// Enum for parameter {{paramName}}
+    ///
+    public enum {{{vendorExtensions.x-swift-nested-enum-type}}}: {{{baseType}}}, Codable, CaseIterable {
+        {{#allowableValues}}
+        {{#enumVars}}
+        case {{name}} = {{{value}}}
+        {{/enumVars}}
+        {{/allowableValues}}
+    }
+    {{/isEnum}}
+    {{/allParams}}
+    {{#vendorExtensions.x-swift-custom-error-type}}
+    public enum {{{vendorExtensions.x-swift-custom-error-type}}}: Error, CustomStringConvertible {
+        {{#responses}}
+        {{#vendorExtensions.x-swift-has-custom-error-type}}
+        // {{{message}}}
+        case code{{{code}}}Error{{#dataType}}({{{dataType}}}){{/dataType}}
+        {{/vendorExtensions.x-swift-has-custom-error-type}}
+        {{/responses}}
+
+        public var description: String {
+            switch self {
+            {{#responses}}
+            {{#vendorExtensions.x-swift-has-custom-error-type}}
+            case .code{{{code}}}Error{{#dataType}}(let object){{/dataType}}:
+                return "{{{vendorExtensions.x-swift-custom-error-type}}}: {{{message}}}{{#dataType}}: \(object){{/dataType}}"
+            {{/vendorExtensions.x-swift-has-custom-error-type}}
+            {{/responses}}
+            }
+        }
+    }
+    {{/vendorExtensions.x-swift-custom-error-type}}
+
+    {{#summary}}
+    /// {{{.}}}
+    {{/summary}}
+    /// - {{httpMethod}} {{{path}}}{{#notes}}
+    /// - {{{.}}}{{/notes}}{{#subresourceOperation}}
+    /// - subresourceOperation: {{.}}{{/subresourceOperation}}{{#defaultResponse}}
+    /// - defaultResponse: {{.}}{{/defaultResponse}}
+    {{#authMethods}}
+    /// - {{#isBasic}}BASIC{{/isBasic}}{{#isOAuth}}OAuth{{/isOAuth}}{{#isApiKey}}API Key{{/isApiKey}}:
```

Comment:
This is likely to create issues because isBasic is also true for HTTP Bearer auth and HTTP Signature auth.
Use isBasicBasic instead